### ISSUE TITLE
Error when `automation_id` field is missing

### ISSUE THREAD

**Issue by mrks115** [gurock/trcli#88]
### TestRail CLI Version

1.2.0

### CLI Environment

Mac M1, Python 3.10.5

### TestRail Version

v7.5.4

### TestRail Instance Type

NA

### Current behavior

Precondition: No custom field `automation_id` in Testrail.

When I upload test results I got the following error:
```
Traceback (most recent call last):
  File "/opt/homebrew/bin/trcli", line 8, in <module>
    sys.exit(cli())
  File "/opt/homebrew/lib/python3.10/site-packages/click/core.py", line 1128, in __call__
    return self.main(*args, **kwargs)
  File "/opt/homebrew/lib/python3.10/site-packages/click/core.py", line 1053, in main
    rv = self.invoke(ctx)
  File "/opt/homebrew/lib/python3.10/site-packages/click/core.py", line 1659, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/opt/homebrew/lib/python3.10/site-packages/click/core.py", line 1395, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/opt/homebrew/lib/python3.10/site-packages/click/core.py", line 754, in invoke
    return __callback(*args, **kwargs)
  File "/opt/homebrew/lib/python3.10/site-packages/click/decorators.py", line 26, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/opt/homebrew/lib/python3.10/site-packages/click/decorators.py", line 84, in new_func
    return ctx.invoke(f, obj, *args, **kwargs)
  File "/opt/homebrew/lib/python3.10/site-packages/click/core.py", line 754, in invoke
    return __callback(*args, **kwargs)
  File "/opt/homebrew/lib/python3.10/site-packages/trcli/commands/cmd_parse_junit.py", line 46, in cli
    result_uploader.upload_results()
  File "/opt/homebrew/lib/python3.10/site-packages/trcli/api/results_uploader.py", line 92, in upload_results
    added_test_cases, result_code = self.add_missing_test_cases(
  File "/opt/homebrew/lib/python3.10/site-packages/trcli/api/results_uploader.py", line 288, in add_missing_test_cases
    ) = self.api_request_handler.check_missing_test_cases_ids(project_id)
  File "/opt/homebrew/lib/python3.10/site-packages/trcli/api/api_request_handler.py", line 295, in check_missing_test_cases_ids
    aut_case_id = case["custom_automation_id"]
KeyError: 'custom_automation_id'
```

### Desired behavior

Expected a catched error (with user error message) instead of a Python error message.

**Comment by BittahCriminal** [comment]
Not sure if this would be a bug, but a feature to fail on automation_id not present. There are times when you have manual tests that haven't been automated yet that you would want to record in testrail.

**Comment by Testinator-X** [comment]
Did you configure the automation_id field in your custom fields as described in the readme?


To allow for this, you must first add a new [custom field](https://www.gurock.com/testrail/docs/user-guide/howto/fields/) of type String with system name automation_id.

**Comment by mrks115** [comment]
No, as described in the issue I did not configure the automation_id field.
> Precondition: No custom field automation_id in Testrail.

But that's the point of this issue:
I'd expect an error message which leads me to the missing automation_id and not Python error.

**Comment by d-rede** [comment]
Thanks for having brought this to our attention. 

This problem should be resolved in v1.4.2 of the TestRail CLI, but please feel free to reopen the issue if you still run into it.
